src/main/kotlin/com/milaboratory/mixcr/cli/CommandBuildLibrary.kt

```diff
@@ -0,0 +1,394 @@
+
+    @set:Option(
+        names = ["--chain"],
+        description = ["Immunological chain"],
+        paramLabel = "<chain>",
+        required = true
+    )
+    var chain: String = ""
+        set(value) {
+            ValidationException.requireKnownChains(value)
+            field = value
+        }
+
+    @set:Option(
+        names = ["--do-not-infer-points"],
+        description = ["Do not infer reference points"],
+        required = false
+    )
+    var doNotInferPoints: Boolean = false
+
+    @set:Option(
+        names = ["--keep-intermediate"],
+        description = ["Keep intermediate files"],
+        required = false
+    )
+    var keepIntermediateFiles: Boolean = false
+
+    @Parameters(
+        description = ["Output library."],
+        paramLabel = "library.json",
+        index = "0",
+    )
+    var output: Path? = null
+        set(value) {
+            ValidationException.requireFileType(value, InputFileType.JSON, InputFileType.JSON_GZ)
+        }
+
+    override val inputFiles: List<Path> get() = emptyList()
+    override val outputFiles: List<Path> get() = listOfNotNull(output)
+
+    private fun mkLibraryForGeneType(gt: GeneType) = run {
+        val input = when (gt) {
+            Variable -> vGenes
+            Diversity -> dGenes
+            Joining -> jGenes
+            Constant -> cGenes
+        }
+        if (input == null) {
+            null
+        } else if (input.species != null) {
+            fromKnownSpecies(gt, input.species!!)
+        } else {
+            val fasta = input.fasta!!
+            fromFasta(gt, fasta.geneFeatureOps!!, fasta.path)
+        }
+    }
+
+    private fun tmpLibraryName(geneType: GeneType) =
+        output!!.absolutePathString().replace(".json", ".${geneType.letter}.json")
+
+    private fun fromKnownSpecies(geneType: GeneType, species: String) = run {
+        val chains = Chains.parse(chain)
+        val data = VDJCLibraryRegistry.getDefaultLibrary(species).data
+        val dataFiltered =
+            data.copy(
+                genes = data.genes.filter { it.geneType == geneType && it.chains.intersects(chains) },
+                speciesNames = listOf(this.species),
+                taxonId = java.lang.Long.parseLong(taxonId)
+            )
+        val libName = tmpLibraryName(geneType)
+        writeToFile(listOf(dataFiltered), libName, false)
+        libName
+    }
+
+    private fun logCmd(cmd: List<String>) {
+        logger.debug { "Running:\nrepseqio ${cmd.joinToString(" ")}" }
+    }
+
+    private fun fromFasta(
+        geneType: GeneType,
+        geneFeature: String,
+        fasta: Path
+    ) = run {
+        val libName = tmpLibraryName(geneType)
+
+        val ffCmd = """
+            fromFasta
+            --taxon-id $taxonId
+            --species-name $species
+            --gene-type ${geneType.letter}
+            --chain $chain
+            --gene-feature $geneFeature
+            --name-index 0
+            -f
```

Comment:
may be inherit it from main command?

Reply:
in other command too

Reply:
do not understand this suggestion

Reply:
you can propogate `-f` from `buildLibrary` to repseqio commands

Reply:
Let's force -f under the hood.